src/foam/util/NUIDGenerator.js

```diff
@@ -78,11 +80,9 @@ foam.CLASS({
       `,
       javaCode: `
         // At least 2 bits sequence number
-        if ( seqNo_.get() == 0 ) {
-          synchronized ( this ) {
-            if ( seqNo_.get() == 0 ) {
-              initMaxSeqNo();
-            }
+        synchronized ( this ) {
+          if ( ! initMaxSeqNo_.get() ) {
```

Comment:
Initialization will occur once so a check before the synchronized block will avoid the synchronized call during normal operation.